File: src/main/java/fit/wenchao/db/Table.java
package fit.wenchao.db;

import java.util.Iterator;
import java.util.List;

public class Table implements Iterable<TableAttr> {
    String name;

    List<TableAttr> attrs;

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    @Override
    public Iterator<TableAttr> iterator() {
        return attrs.iterator();
    }

    public void setAttrs(List<TableAttr> attrs) {
        this.attrs = attrs;
    }

    public JavaSourceFile generateJavaModel(JavaPackage javaPackage) {
        JavaSourceFile javaSourceFile = JavaSourceFile.ofMysqlModel(this, javaPackage);
        //System.out.println(javaSourceFile);
        if (!javaSourceFile.existsIn(javaPackage)) {
            javaSourceFile.put2Package(javaPackage);
        }
        return javaSourceFile;
    }

    public void generateMybatisMapper(JavaPackage javaPackage, JavaSourceFile poSourceFile) {
        JavaSourceFile javaSourceFile = JavaSourceFile.ofMybatisMapper(this, javaPackage, poSourceFile);
        //System.out.println(javaSourceFile);
        if (!javaSourceFile.existsIn(javaPackage)) {
            javaSourceFile.put2Package(javaPackage);
        }
    }

    public JavaSourceFile generateDao(JavaPackage javaPackage) {
        JavaSourceFile javaSourceFile = JavaSourceFile.ofDao(this, javaPackage);
        //System.out.println(javaSourceFile);
        if (!javaSourceFile.existsIn(javaPackage)) {
            javaSourceFile.put2Package(javaPackage);
        }
        return javaSourceFile;
    }

    public void generateDaoImpl(JavaPackage javaPackage, JavaSourceFile daoSourceFile) {
        JavaSourceFile javaSourceFile = JavaSourceFile.ofDaoImpl(this, javaPackage, daoSourceFile);
        //System.out.println(javaSourceFile);
        if (!javaSourceFile.existsIn(javaPackage)) {
            javaSourceFile.put2Package(javaPackage);
        }
    }

    public void generateService(JavaPackage javaPackage) {
        JavaSourceFile javaSourceFile = JavaSourceFile.ofService(this, javaPackage);
        if (!javaSourceFile.existsIn(javaPackage)) {
            javaSourceFile.put2Package(javaPackage);
        }
    }

    public void generateServiceImpl(JavaPackage javaPackage, JavaSourceFile serviceSourceFile) {
        JavaSourceFile javaSourceFile = JavaSourceFile.ofServiceImpl(this, javaPackage, serviceSourceFile);
        //System.out.println(javaSourceFile);
        if (!javaSourceFile.existsIn(javaPackage)) {
            javaSourceFile.put2Package(javaPackage);
        }
    }
}

File: src/main/kotlin/fit/wenchao/db/Table.kt
package fit.wenchao.db

import fit.wenchao.db.constants.Lang

class Table : Iterable<TableAttr> {

    var name: String = ""

    var attrs: List<TableAttr> = mutableListOf()

    override fun iterator(): Iterator<TableAttr> {
        return attrs.iterator()
    }

    fun generateJavaModel(javaPackage: JavaPackage): JavaSourceFile {
        val javaSourceFile = ofMysqlModel(this, javaPackage, Lang.KOTLIN)
        if (javaSourceFile.notExistsIn(javaPackage)) {
            javaSourceFile.put2Package(javaPackage)
        }
        return javaSourceFile
    }

    fun generateMybatisMapper(javaPackage: JavaPackage, poSourceFile: JavaSourceFile): JavaSourceFile {
        val javaSourceFile = ofMybatisMapper(this, javaPackage, poSourceFile, Lang.KOTLIN)
        if (javaSourceFile.notExistsIn(javaPackage)) {
            javaSourceFile.put2Package(javaPackage)
        }
        return javaSourceFile
    }

    fun generateDao(javaPackage: JavaPackage, poJavaSourceFile: JavaSourceFile): JavaSourceFile {
        val javaSourceFile = ofDao(this, javaPackage, poJavaSourceFile, Lang.KOTLIN)
        // System.out.println(javaSourceFile);
        if (javaSourceFile.notExistsIn(javaPackage)) {
            javaSourceFile.put2Package(javaPackage)
        }
        return javaSourceFile
    }

    fun generateDaoImpl(
        javaPackage: JavaPackage,
        mapperSourceFile: JavaSourceFile,
        poSourceFile: JavaSourceFile,
        daoSourceFile: JavaSourceFile
    ) {
        val javaSourceFile = ofDaoImpl(
            this,
            javaPackage, mapperSourceFile, poSourceFile, daoSourceFile, Lang.KOTLIN
        )
        if (javaSourceFile.notExistsIn(javaPackage)) {
            javaSourceFile.put2Package(javaPackage)
        }
    }

    fun generateService(javaPackage: JavaPackage): JavaSourceFile {
        val javaSourceFile = ofService(this, javaPackage, Lang.KOTLIN)
        if (javaSourceFile.notExistsIn(javaPackage)) {
            javaSourceFile.put2Package(javaPackage)
        }
        return javaSourceFile
    }

    fun generateServiceImpl(javaPackage: JavaPackage, serviceSourceFile: JavaSourceFile) {
        val javaSourceFile = ofServiceImpl(this, javaPackage, serviceSourceFile, Lang.KOTLIN)
        if (javaSourceFile.notExistsIn(javaPackage)) {
            javaSourceFile.put2Package(javaPackage)
        }
    }
}